packages/idos-sdk-js/src/lib/auth.ts

```diff
@@ -171,8 +177,24 @@ export class Auth {
       );
     };
 
+    const { connect } = nearAPI;
+    const connectionConfig = {
+      networkId: import.meta.env.VITE_IDOS_NEAR_DEFAULT_NETWORK,
+      nodeUrl: import.meta.env.VITE_IDOS_NEAR_DEFAULT_RPC_URL,
+    };
+    const nearConnection = await connect(connectionConfig);
+
+    const response = await nearConnection.connection.provider.query({
+      request_type: "view_access_key_list",
+      finality: "final",
+      account_id: currentAddress,
+    });
+    const public_key_with_prefix = response.keys.find((element: object) => element.access_key.permission == "FullAccess").public_key;
+    const public_key = public_key_with_prefix.replace(/^ed25519:/, "");
```

Comment:
Non-actionable warning: not all NEAR keys are going to be `ed25519`. They also used to support secpk keys, but I'm seeing support for those being phased out (probably over design decision on their side with the EVM compatibility layer they're introducing).

See https://github.com/near/near-sdk-rs/blob/1408127feefb37e7f84eba25984af18af46cd91d/near-sdk/src/types/public_key.rs#L228